### ISSUE TITLE
Add type annotations support for optional properties

### ISSUE THREAD

**Issue by olegshtch** [vojtechhabarta/typescript-generator#412]
Checker framework uses type annotations with `@Target(value = {ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})` so beans looks like:

```java
class Bean {
 private @Nullable String prop1;
 public @Nullable String getProp1() {...};
 public void setProp1(@Nullable String) {...};
}
```

But the generator can not extract those annotations and generated code misses optional information.

**Comment by vojtechhabarta** [comment]
There are following parameters which affect optionality: [optionalProperties](http://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#optionalProperties), [optionalAnnotations](http://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#optionalAnnotations) and [optionalPropertiesDeclaration](http://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#optionalPropertiesDeclaration).

Here is Maven sample for your case:
``` xml
<configuration>
    <optionalProperties>useSpecifiedAnnotations</optionalProperties>
    <optionalAnnotations>
        <annotation>javax.annotation.Nullable</annotation>
    </optionalAnnotations>
    ...
</configuration>
```

**Comment by olegshtch** [comment]
Yes, it works with `ElementType.ANNOTATION_TYPE` like `javax.annotation.Nullable` but doesn't work with `ElementType.TYPE_USE`, `ElementType.TYPE_PARAMETER` annotations.

**Comment by vojtechhabarta** [comment]
Is it `org.checkerframework.checker.nullness.qual.Nullable`? This annotation is not accessible in runtime despite it has `@Retention(RetentionPolicy.RUNTIME)` (which is surprising to me).

See following test:
``` java
public class TestCheckerNullableAnnotation {

    public @javax.annotation.Nullable String javaxProp;
    public @org.checkerframework.checker.nullness.qual.Nullable String checkerProp;

    public static void main(String[] args) throws Exception {

        // prints "@javax.annotation.Nullable()"
        System.out.println(TestCheckerNullableAnnotation.class.getField("javaxProp")
                .getAnnotation(javax.annotation.Nullable.class));

        // prints "null"
        System.out.println(TestCheckerNullableAnnotation.class.getField("checkerProp")
                .getAnnotation(org.checkerframework.checker.nullness.qual.Nullable.class));
    }

}
```
So typescript-generator is not able to get this annotation.

**Comment by olegshtch** [comment]
For a type annotation it looks like:
```java
System.out.println(TestCheckerNullableAnnotation.class.getField("checkerProp").getAnnotatedType()
                .getAnnotation(org.checkerframework.checker.nullness.qual.Nullable.class))
```

**Comment by vojtechhabarta** [comment]
Yes, that works, thanks.
I will add support for these annotations.

**Comment by vojtechhabarta** [comment]
Released in version [2.17.558](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.17.558).
